src/Data/Fin/Properties.agda

```diff
@@ -26,49 +26,48 @@ open import Relation.Binary.PropositionalEquality as P
 open import Category.Functor
 open import Category.Applicative
 
-open DecTotalOrder N.decTotalOrder using () renaming (refl to ℕ≤-refl)
+open DecTotalOrder N.decTotalOrder using () renaming (
+  refl to ℕ≤-refl
+  ; trans to ℕ≤-trans
+  ; antisym to ℕ≤-antisym
+  ; total to ℕ≤-total
+  ; _≤?_ to _ℕ≤?_)
 
 ------------------------------------------------------------------------
--- Properties
+-- Equality properties
+
+infix 4 _≟_
 
 suc-injective : ∀ {o} {m n : Fin o} → Fin.suc m ≡ suc n → m ≡ n
 suc-injective refl = refl
 
+_≟_ : {n : ℕ} → Decidable {A = Fin n} _≡_
+zero  ≟ zero    = yes refl
+zero  ≟ suc y = no λ()
+suc x ≟ zero    = no λ()
+suc x ≟ suc y with x ≟ y
+... | yes x≡y = yes (cong suc x≡y)
+... | no  x≢y = no (x≢y ∘ suc-injective)
+
 preorder : ℕ → Preorder _ _ _
 preorder n = P.preorder (Fin n)
 
 setoid : ℕ → Setoid _ _
 setoid n = P.setoid (Fin n)
 
-cmp : ∀ {n} → Trichotomous _≡_ (_<_ {n})
-cmp zero    zero    = tri≈ (λ())     refl  (λ())
-cmp zero    (suc j) = tri< (s≤s z≤n) (λ()) (λ())
-cmp (suc i) zero    = tri> (λ())     (λ()) (s≤s z≤n)
-cmp (suc i) (suc j) with cmp i j
-... | tri<  lt ¬eq ¬gt = tri< (s≤s lt)         (¬eq ∘ suc-injective) (¬gt ∘ N.≤-pred)
-... | tri> ¬lt ¬eq  gt = tri> (¬lt ∘ N.≤-pred) (¬eq ∘ suc-injective) (s≤s gt)
-... | tri≈ ¬lt  eq ¬gt = tri≈ (¬lt ∘ N.≤-pred) (cong suc eq)    (¬gt ∘ N.≤-pred)
-
-strictTotalOrder : ℕ → StrictTotalOrder _ _ _
-strictTotalOrder n = record
-  { Carrier            = Fin n
-  ; _≈_                = _≡_
-  ; _<_                = _<_
-  ; isStrictTotalOrder = record
-    { isEquivalence = P.isEquivalence
-    ; trans         = N.<-trans
-    ; compare       = cmp
-    }
-  }
-
+isDecEquivalence : ∀ {n} → IsDecEquivalence (_≡_ {A = Fin n})
+isDecEquivalence = record {
```

Comment:
Please adhere to the style of list formatting of the standard library.  
For records, braces and semicolons are aligned, `=`s also.